src/main/scala/euler/archives/page/one/problemOne.scala

```diff
@@ -0,0 +1,217 @@
+    var check = 0;
+    for{row <- grid.indices
+        column <- grid.indices} {
+      // while on the current column we want to go forward by three row+
+
+      if(row +3 < grid.length){
+//        check the product of the grid column and row
+        check = grid(column)(row)* grid(column)(row+1)*grid(column)(row+2)*grid(column)(row+3)
+        if(check>max){
+          max = check
+//          saving the indices for use later
+          r1 = row
+          r2 = row+1
+          r3 = row + 2
+          r4 = row + 3
+          c1 = column
+          c2 = column
+          c3 = column
+          c4 = column
+        }
+      }
+//      while on the current row we want to go three column forward
+      if(column + 3 < grid.length){
+//        check the product of the grid column and rows
+        check = grid(column)(row) * grid(column+1)(row)* grid(column+2)(row) * grid(column+3)(row)
+        if(check > max){
+            max = check
+          //          saving the indices for use later
+          r1 = row
+          r2 = row
+          r3 = row
+          r4 = row
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+
+//      we want to go diagonally for row and column, meaning as we move one row, we would move one column, this would produce a diagonal like structure
+      if((row +3  < grid.length) && (column +3 < grid.length)){
+        //        check the product along the diagonal downward
+
+        check = grid(column)(row) * grid(column+1)(row+1)* grid(column+2)(row+2) * grid(column+3)(row+3)
+        if(check > max){
+          max = check
+          r1 = row
+          r2 = row + 1
+          r3 = row + 2
+          r4 = row + 3
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+    // we want to go diagonally for row and column, meaning as we move one row,
+    // we would move one column, this would produce a diagonal like structure but this time around upward
+      if(row > 3 && (column +3 < grid.length)){
+        //        check the product along the diagonal upward
+        check =  grid(column)(row) * grid(column +1)(row -1) * grid(column+2)(row-2) * grid(column +3)(row-3)
+        if(check > max ){
+          max = check
+          r1 = row
+          r2 = row - 1
+          r3 = row - 2
+          r4 = row - 3
+          c1 = column
+          c2 = column +1
+          c3 = column + 2
+          c4 = column + 3
+        }
+      }
+
+    }
+
+
+    max
+  }
+
+  def totalNameScores:Int = {
+    import scala.io.Source
+    val bufferedSource = Source.fromFile("p022_names.txt")
+    var text: String = ""
+    for (line <- bufferedSource.getLines()) {
+      text += line
+    }
+    bufferedSource.close
+
+    val names = text.split(",").toSeq
+    val sum = names.filter(_.nonEmpty).map(_.trim.toLowerCase()).sorted.zipWithIndex
```

Comment:
is `sorted` needed here?

Reply:
Yes it kind of needed base on the question.  I need to get the position of a word after sorting in ASC order